dynamodb/src/main/scala/zio/dynamodb/DynamoDBExecutorImpl.scala

```diff
@@ -0,0 +1,601 @@
+      case BatchWriteItem.Put(item)   =>
+        WriteRequest(Some(PutRequest(item.map.map { case (k, v) => (k, buildAwsAttributeValue(v)) })), None)
+    }
+
+  private def doUpdateItem(updateItem: UpdateItem): ZIO[Any, Throwable, Option[Item]] =
+    dynamoDb.updateItem(generateUpdateItemRequest(updateItem)).mapBoth(_.toThrowable, optionalItem)
+
+  /* REVIEW(avi / john): We get an empty map back here if we pass a None to returned Items, but we can also get an empty map
+        if we haven't updated anything. Do we want to return an empty map if we haven't updated anything or return a None?
+        I feel like a None makes more sense here but want your thoughts as well.
+   */
+  private def optionalItem(updateItemResponse: UpdateItemResponse.ReadOnly): Option[Item] =
+    updateItemResponse.attributesValue.flatMap(m => toOption(m).map(toDynamoItem))
+
+  private def doScanSome(scanSome: ScanSome): ZIO[Any, Throwable, (Chunk[Item], LastEvaluatedKey)] = {
+    val stream =
+      dynamoDb
+        .scan(generateScanRequest(scanSome))
+        .mapBoth(_.toThrowable, toDynamoItem)
+    stream.run(ZSink.collectAll[Item]).map(chunk => (chunk, chunk.lastOption))
+  }
+
+  private def generateUpdateItemRequest(updateItem: UpdateItem): UpdateItemRequest = {
+    val maybeUpdateConditionExpr =
+      zipOptionalRenderables(Some(updateItem.updateExpression), updateItem.conditionExpression)
+
+    UpdateItemRequest(
+      tableName = updateItem.tableName.value,
+      key = updateItem.key.map.map { case (k, v) => (k, buildAwsAttributeValue(v)) },
+      returnValues = Some(buildAwsReturnValue(updateItem.returnValues)),
+      returnConsumedCapacity = Some(buildAwsReturnConsumedCapacity(updateItem.capacity)),
+      returnItemCollectionMetrics = Some(buildAwsItemMetrics(updateItem.itemMetrics)),
+      updateExpression = maybeUpdateConditionExpr.flatMap(_._2._1),
+      expressionAttributeValues =
+        maybeUpdateConditionExpr.flatMap(m => aliasMapToExpressionZIOAwsAttributeValues(m._1)),
+      conditionExpression = maybeUpdateConditionExpr.flatMap(_._2._2)
+    )
+  }
+
+  private def doScanAll(scanAll: ScanAll): ZIO[Any, Throwable, Stream[Throwable, Item]] =
+    ZIO.succeed(
+      dynamoDb
+        .scan(generateScanRequest(scanAll))
+        .mapBoth(
+          awsError => awsError.toThrowable,
+          item => toDynamoItem(item)
+        )
+    )
+
+  private def generateScanRequest(scanSome: ScanSome): ScanRequest = {
+    val filterExpression = scanSome.filterExpression.map(fe => fe.render.render(AliasMap.empty))
+    ScanRequest(
+      tableName = scanSome.tableName.value,
+      indexName = scanSome.indexName.map(_.value),
+      select = scanSome.select.map(buildAwsSelect),
+      exclusiveStartKey = scanSome.exclusiveStartKey.map(m => attrMapToAwsAttrMap(m.map)),
+      returnConsumedCapacity = Some(buildAwsReturnConsumedCapacity(scanSome.capacity)),
+      // Looks like we're not currently supporting segments?
```

Comment:
yes we have an issue for this enhancement https://github.com/zio/zio-dynamodb/issues/43